server/chat-plugins/mafia.ts

```diff
@@ -383,6 +362,30 @@ class MafiaTracker extends Rooms.RoomGame {
 		}
 	}
 
+	static isGameBanned(room: Room, user: User) {
+		return Punishments.getRoomPunishType(room, toID(user)) === 'MAFIAGAMEBAN';
+	}
+
+	static gameBan(room: Room, user: User, reason: string, duration: number) {
+		Punishments.roomPunish(room, user, ['MAFIAGAMEBAN', toID(user), Date.now() + (duration * 24 * 60 * 60 * 1000), reason]);
+	}
+
+	static ungameBan(room: Room, user: User) {
+		Punishments.roomUnpunish(room, toID(user), 'MAFIAGAMEBAN', false);
+	}
+
+	static isHostBanned(room: Room, user: User) {
+		return Punishments.getRoomPunishType(room, toID(user)) === 'MAFIAHOSTBAN' || this.isGameBanned(room, user);
```

Comment:
nit, you should probably just get the punishment once and check it against hostban/gameban in this function